### ISSUE TITLE
Problem with make_standalond_toolchain.py

### ISSUE THREAD

**Issue by robiwano** [android-ndk/ndk#204]
With the standalone toolchain from ndk r12b:
`./build/tools/make_standalone_toolchain.py --api 21 --arch arm --stl libc++ --force --install-dir standalone/arm`

the following code fails:

```c++
#include <assert.h>
#include <cmath>
#include <iostream>

int main(int, const char* [])
{
    auto value = std::log2(17.0);
    std::cout << "value = " << value << std::endl;
    return 0;
}
```

compiled with `"standalone\arm\bin\clang++.cmd" -std=c++11 -pie -static-libstdc++ main.cpp` yields following erroneous result:
`value = -1077`

whilst compiling with non-standalone it seems to yield correct values. Ideas?

**Comment by robiwano** [comment]
Same code works with NDK r11c standalone toolchain.

**Comment by DanAlbert** [comment]
Just confirmed that this is still a problem with r13-beta2, and that this also affects the shared library variant.

If I had to guess, I'd say this is similar to https://code.google.com/p/android/issues/detail?id=212634, but I haven't really dug in yet.

**Comment by DanAlbert** [comment]
Well, that was a fun one. Turns out the problem is that libandroid_support's (the library that backs libc++ to support old android versions) had a completely broken implementation of `log2` for armeabi (armeabi-v7a was fine).

There's a bug in Clang that means that standalone toolchains will always load the arm5 version of a library, not the arm7 one: https://github.com/android-ndk/ndk/issues/151#issuecomment-231209052 (Googlers, http://b/26180518 is the first investigation of this bug).

This didn't show up in ndk-build because ndk-build doesn't rely on the compiler being able to find libraries; it passes the full path to the library to the linker.

I haven't dug in to why this worked in r11. We did change the way libc++/libandroid_support were built and linked in r12, so my guess is that the libandroid_support definition simply wasn't getting used in r11.

Anyway, the fix is now up for review: https://android-review.googlesource.com/#/c/276095/

Since this fix is trivially correct and non-invasive, I'm going to cherry-pick this to r13.
